ipv6/uint128.go

```diff
@@ -0,0 +1,207 @@
+/*
+This file takes influence from uint128: https://github.com/davidminor/uint128
+
+Copyright (c) 2014 David Minor
+
+Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the "Software"), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+*/
+
+package ipv6
+
+import "math/bits"
+
+type uint128 struct {
+	high, low uint64
+}
+
+// Uint128FromUint64 returns a uint128 from its two 64 bit unsigned representations
+func Uint128FromUint64(high uint64, low uint64) uint128 {
```

Comment:
Just to reduce the amount of code we put into this, make sure that this type doesn't have anything we don't need. For example, I don't see this called anywhere in this PR so it should be removed.

Even if we need it later for implementing masks or trie nodes or something leave it out of this PR and then add it when it comes up.

My reasoning for this is that uint128 isn't a product of this library. So, it doesn't need to be complete and I'd rather not carry code that isn't tested as part of the final product.